codex-rs/login/src/server.rs

```diff
@@ -198,24 +251,59 @@ pub fn run_login_server(
                     }
                     let _ = req.respond(resp);
                     shutdown_flag.store(true, Ordering::SeqCst);
+
+                    // Signal completion to timeout watcher.
+                    let _ = done_tx.send(());
                     return Ok(());
                 }
                 _ => {
                     let _ = req.respond(Response::from_string("Not Found").with_status_code(404));
                 }
             }
         }
-        Err(io::Error::other("Login flow was not completed"))
+
+        // Signal completion to timeout watcher.
+        let _ = done_tx.send(());
```

Comment:
Under what conditions have we reached this point? It's always when `shutdown_flag` is `true`, isn't it? So why are we updating the timeout watcher here: to cancel it?